lib/thrift/generator/struct_generator.ex

```diff
@@ -63,22 +57,37 @@ defmodule Thrift.Generator.StructGenerator do
         def new, do: %__MODULE__{}
         unquote_splicing(List.wrap(extra_defs))
 
-        defmodule BinaryProtocol do
-          @moduledoc false
-          unquote_splicing(binary_protocol_defs)
+        defprotocol SerDe do
```

Comment:
Should we introduce a `defserde` macro in `Thrift.Protocol` that generates this `SerDe` protocol? We could use it here and also where we define `TApplicationException.SerDe`. It would be used by additional protocol implementations, too.

Reply:
Good idea.